### PR TITLE
fix: focus monitor cleanup

### LINKED ISSUE #6555: Focus Monitor detached DOM Nodes leak
### Reproduction link
[https://codesandbox.io/s/flamboyant-moore-llt8t?file=/src/app/app.component.ts](https://codesandbox.io/s/flamboyant-moore-llt8t?file=/src/app/app.component.ts)

### Steps to reproduce
1. Open Performance Monitor inside dev tools
2. Use the toggle button a few times
3. Notice the increase in DOM Nodes count
4. Try to force the garbage collector

### What is expected?
After each toggle the DOM Nodes should be cleared as they are no longer visible and should be dropped and it should be possible to clear them forcing the garbage collector.

### What is actually happening?
The DOM Nodes stays in the memory as detached HTMLElements and it's clearly visible inside the heap snapshot with the reference to `_elementInfo` inside FocusMonitor.

| Environment | Info |
|---|---|
| ng-zorro-antd | 11.3.0 |
| Browser | Google Chrome 89.0.4389.90 |

---
After a quick look inside the code I think what's missing is `stopMonitoring` method being used inside `ngOnDestroy`.
I think it got a bit messy with the focus monitor cleanup across various components for e.g checkbox uses the `stopMonitoring` method but doesn't clean up after monitor method subscription where input group clears after the subscription but doesn't run `stopMonitoring` method.

### PR DESCRIPTION
**Pull request by raizemm**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: close #6555 

## Does this PR introduce a breaking change?
```
[ ] Yes
[n] No
```

### CODE REVIEW

File: components/checkbox/checkbox-group.component.ts [hunk 1]
```diff
@@ -79,11 +79,13 @@ export class NzCheckboxGroupComponent implements ControlValueAccessor, OnInit, O
   }
 
   ngOnInit(): void {
-    this.focusMonitor.monitor(this.elementRef, true).subscribe(focusOrigin => {
-      if (!focusOrigin) {
-        Promise.resolve().then(() => this.onTouched());
-      }
-    });
+    this.focusMonitor.monitor(this.elementRef, true)
+      .pipe(takeUntil(this.destroy$))
```
Comment (hsuanxyz) [inline components/checkbox/checkbox-group.component.ts:83]: Thanks for your contribution, I see that just unsubscribing doesn't make it stop monitoring, we need to call `stopMonitoring` explicitly. In this case here we can safely do it, could you modify them?

Comment (raizemm) [reply components/checkbox/checkbox-group.component.ts:83]: Hi @hsuanxyz, as far as I can see there is already a `stopMonitoring` call inside `ngOnDestroy` of this component. 
Some of the components were calling `stopMonitoring` but didn't `unsubscribe` from the stream.

Comment (hsuanxyz) [reply components/checkbox/checkbox-group.component.ts:83]: Oh, I got. @raizemm  would you plz rebase this pr onto master.

Comment (raizemm) [reply components/checkbox/checkbox-group.component.ts:83]: Done. @hsuanxyz

**Review by hsuanxyz — APPROVED** [review]
LGTM

**Pull request merged**